netket/optimizer/sr/sr_onthefly_logic.py

```diff
@@ -40,6 +40,14 @@ def O_vjp(forward_fn, params, samples, w):
     return jax.tree_map(sum_inplace, res)  # allreduce w/ MPI.SUM
 
 
+def O_vjp_rc(forward_fn, params, samples, w):
+    _, vjp_fun = jax.vjp(forward_fn, params, samples)
+    res_r, _ = vjp_fun(w)
+    res_i, _ = vjp_fun(-1.0j * w)
```

Comment:
I think it might be more efficient to relax the real parameters to complex and do just 1 backward pass, however its giving me errors in some cases. I suggest we merge this as is and maybe revisit it later on.

Reply:
Interesting.
To be discussed